common/lib/xmodule/xmodule/modulestore/tests/django_utils.py

```diff
@@ -198,6 +198,49 @@ def xml_store_config(data_dir, source_dirs=None):
 )
 
 
+class ModuleStoreClassCleanupTestCase(TestCase):
+    """
+    Sometimes you want to create a course and know it will stick around between
+    setUp()s.
+
+    Note that if you create any Django ORM data in your setUpClass() methods,
+    you must manually clean it up in your tearDownClass(). Django's built-in
+    TestCase class (which this class inherits from) will cleanup the database
+    between tests on databases that support transactions, but it cannot clean
+    up models created in setUpClass(). In Django 1.8, we will be able to use
+    setUpTestData() for this.
+    """
+    MODULESTORE = mixed_store_config(mkdtemp_clean(), {}, include_xml=False)
+
+    @classmethod
+    def setUpClass(cls):
+        super(ModuleStoreClassCleanupTestCase, cls).setUpClass()
+
+        cls._settings_override = override_settings(MODULESTORE=cls.MODULESTORE)
+        cls._settings_override.__enter__()
+        XMODULE_FACTORY_LOCK.enable()
+        clear_existing_modulestores()
+        cls.store = modulestore()
+
+    @classmethod
+    def tearDownClass(cls):
+        ModuleStoreTestCase.drop_mongo_collections()  # refactor this
+        RequestCache().clear_request_cache()
+        XMODULE_FACTORY_LOCK.disable()
+        cls._settings_override.__exit__(None, None, None)
+
+        super(ModuleStoreClassCleanupTestCase, cls).tearDownClass()
+
+    def setUp(self):
+        # When testing CCX, we should make sure that
```

Comment:
Probably better to make this content less CCX specific, since it's really just about any field overrides.